parlai/agents/image_seq2seq/image_seq2seq.py

```diff
@@ -0,0 +1,349 @@
+                num_enc_embs = min(
+                    self.model.encoder.embeddings.weight.shape[0], enc_embs.shape[0]
+                )
+                num_dec_embs = min(
+                    self.model.decoder.embeddings.weight.shape[0], dec_embs.shape[0]
+                )
+                for i in range(num_embs):
+                    self.model.embeddings.weight.data[i] = embs[i]
+                    cnt += 1
+                for i in range(num_enc_embs):
+                    self.model.encoder.embeddings.weight.data[i] = enc_embs[i]
+                for i in range(num_dec_embs):
+                    self.model.decoder.embeddings.weight.data[i] = dec_embs[i]
+                state_dict['embeddings.weight'] = self.model.embeddings.weight
+                state_dict[
+                    'encoder.embeddings.weight'
+                ] = self.model.encoder.embeddings.weight
+                state_dict[
+                    'decoder.embeddings.weight'
+                ] = self.model.decoder.embeddings.weight
+
+                print(
+                    'Initialized embeddings for {} tokens ({}%).'
+                    ''.format(cnt, round(cnt * 100 / len(self.dict), 1))
+                )
+        self.model.load_state_dict(state_dict)
+
+    def eval_step(self, batch: Batch) -> Output:
+        """
+        Evaluate a single batch of examples.
+
+        Override to compute correct BLEU scores.
+        """
+        output = super().eval_step(batch)
+        if output is None or (self.skip_generation and not self.compute_tokenized_bleu):
+            return output
+
+        texts = output.text
+        self._compute_fairseq_bleu(batch, texts)
+        self._compute_nltk_bleu(batch, texts)
+
+        return output
+
+    def report(self) -> dict:
+        """
+        Override to include custom BLEU computations.
+        """
+        metrics = super().report()
+        metrics.update({'fairseq_bleu': 'N/A', 'nltk_bleu_unnormalized': 'N/A'})
+        if not self.skip_generation and self.compute_tokenized_bleu:
+            if fairseq_bleu is not None:
+                try:
+                    fairseq_bleu_scores = {
+                        k: self.fairseq_bleu_scorer.result_string(order=k)
+                        for k in range(1, 5)
+                    }
+                except ZeroDivisionError:
+                    # some preds are REAL bad
+                    fairseq_bleu_scores = {k: 0 for k in range(1, 5)}
+
+                metrics['fairseq_bleu'] = {
+                    k: v[v.index('= ') : v.index(',')]
+                    for k, v in fairseq_bleu_scores.items()
+                }
+            if nltkbleu is not None:
+                metrics['nltk_bleu_unnormalized'] = {
+                    k: round_sigfigs(v / self.nltk_bleu_cnts[k], 4)
+                    for k, v in self.nltk_bleu.items()
+                }
+        return metrics
+
+    def _compute_fairseq_bleu(self, batch: Batch, texts: List[str]):
```

Comment:
worth bubbling up to TGA?

Reply:
I suppose I could include there, initially i did not want to clutter TGA with this but if it's useful...